src/kinematics_plugin.cpp

```diff
@@ -164,58 +156,23 @@ struct BioIKKinematicsPlugin : kinematics::KinematicsBase {
 
   mutable Problem problem;
 
-  static moveit::core::RobotModelConstPtr
-  loadRobotModel(const std::string &robot_description) {
-    static std::map<std::string, moveit::core::RobotModelConstPtr>
-        robot_model_cache;
-    static std::mutex cache_mutex;
-    std::lock_guard<std::mutex> lock(cache_mutex);
-    if (robot_model_cache.find(robot_description) == robot_model_cache.end()) {
-      rdf_loader::RDFLoader rdf_loader(robot_description);
-      auto srdf = rdf_loader.getSRDF();
-      auto urdf_model = rdf_loader.getURDF();
-
-      if (!urdf_model || !srdf) {
-        LOG("URDF and SRDF must be loaded for kinematics solver to work.");
-        return nullptr;
-      }
-      robot_model_cache[robot_description] = moveit::core::RobotModelConstPtr(
-          new robot_model::RobotModel(urdf_model, srdf));
+  template <class T>
+  void getRosParam(const std::string &param, T &val, const T &default_val)
```

Comment:
What if we kept this named `lookupParam` to minimize the diff?

Reply:
`lookupParam` is a method of the inherited `kinematics::KinematicsBase` class. I'm not using it here since it doesn't give the option of declaring parameters if they don't already exist. Perhaps @tylerjw can provide insight on whether using the inherited `lookupParam` or this `getRosParam` is preferred?

Reply:
lookupParam should probably support declaring parameters. It just hasn't been added in the initial port.